ecosystem/typescript/sdk/src/transactions/async_queue.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * The AsyncQueue class is an async-aware data structure that provides a queue-like
+ * behavior for managing asynchronous tasks or operations.
+ * It allows to enqueue items and dequeue them asynchronously.
+ */
+
```

Comment:
nit line

let's make it clear in the documentation here that this is not thread-safe but it is async concurrency safe and that it doesn't not guarantee ordering for those that call into and await on enqueue.